frontend/src/components/App/App.tsx

```diff
@@ -1,32 +1,40 @@
 import React, { useRef } from 'react'
 import { Route, Routes } from 'react-router-dom'
+import axios from 'axios'
+import Cookies from 'js-cookie'
 import './App.css'
 
 import BikeGallery from '../BikeGallery/BikeGallery'
 import BikeDetails from '../BikeDetails/BikeDetails'
-import Header from '../Header/Header'
+import SignUp from '../SignUp/SignUp'
+import Login from '../Login/Login'
 
-import useBikesApi from '../../hooks/useBikesApi'
-import AddBike from '../AddBike/AddBike'
+// @ts-ignore
+axios.interceptors.request.use(
+   function (config) {
+      return fetch('/api/csrf/').then(() => {
+         config.headers['X-XSRF-TOKEN'] = Cookies.get('XSRF-TOKEN')
+         return config
+      })
+   },
+   function (error) {
+      return Promise.reject(error)
+   }
+)
 
 function App() {
-   const { bikes, addBike, editBike, deleteBike, loading } = useBikesApi()
+    const addBikeInputRef = useRef() as React.MutableRefObject<HTMLInputElement>
```

Comment:
We don't understand this line so we cannot give any input on it 😢

Reply:
[check this](https://beta.reactjs.org/reference/react/useRef)